python/lsst/meas/algorithms/loadReferenceObjects.py

```diff
@@ -0,0 +1,252 @@
+from __future__ import absolute_import, division, print_function
+
+import abc
+
+import numpy
+
+import lsst.afw.table as afwTable
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+
+__all__ = ["getRefFluxField", "getRefFluxKeys", "LoadReferenceObjectsTask", "LoadReferenceObjectsConfig"]
+
+def getRefFluxField(schema, filterName=None):
+    """!Get name of flux field in schema
+
+    if filterName is specified:
+        return <filterName>_camFlux if present
+        else return <filterName>_flux if present (camera filter name matches reference filter name)
+        else throw RuntimeError
+    else:
+        return camFlux, if present,
+        else throw RuntimeError
+
+    @param[in] schema  reference catalog schema
+    @param[in] filterName  name of camera filter
+    @return flux field name
+    @throw RuntimeError if appropriate field is not found
+    """
+    if not isinstance(schema, afwTable.Schema):
+        raise RuntimeError("schema=%s is not a schema" % (schema,))
+    if filterName:
+        fluxFieldList = [filterName + "_camFlux", filterName + "_flux"]
+    else:
+        fluxFieldList = ["camFlux"]
+    for fluxField in fluxFieldList:
+        if fluxField in schema:
+            return fluxField
+
+    raise RuntimeError("Could not find flux field(s) %s" % (", ".join(fluxFieldList)))
+
+def getRefFluxKeys(schema, filterName=None):
+    """!Return flux and flux error keys
+
+    @param[in] schema  reference catalog schema
+    @param[in] filterName  name of camera filter
+    @return a pair of keys:
+        flux key
+        flux error key, if present, else None
+    @throw RuntimeError if flux field not found
+    """
+    fluxField = getRefFluxField(schema, filterName)
+    fluxErrField = fluxField + "Sigma"
+    fluxKey = schema[fluxField].asKey()
+    try:
+        fluxErrKey = schema[fluxErrField].asKey()
+    except Exception:
+        fluxErrKey = None
+    return (fluxKey, fluxErrKey)
+
+class LoadReferenceObjectsConfig(pexConfig.Config):
+    pixelMargin = pexConfig.RangeField(
+        doc = "Padding to add to 4 all edges of the bounding box (pixels)",
+        dtype = int,
+        default = 50,
+        min = 0,
+    )
+    defaultFilter = pexConfig.Field(
+        doc = "Default reference catalog filter to use if filter not specified in exposure; " + \
+            "if blank then filter must be specified in exposure",
```

Comment:
We may have clashed on this before, in which case I'm not going to fight you, but I find:
```
("line 1..... "
 "...line 2")
```
more readable than ` + \` for multi-line strings.  It's also more efficient, I believe, since it makes the concatenation happen during compilation to .pyc, not at runtime, but of course that's not a big deal in these sorts of cases.